sdk/rust/oak/src/lib.rs

```diff
@@ -299,6 +299,146 @@ pub fn node_create_with_label(
     result_from_status(status as i32, ())
 }
 
+/// Get the [`Label`] for the channel associated with the `handle`.
+pub fn channel_label(handle: Handle) -> Result<Label, OakStatus> {
+    let mut bytes = Vec::with_capacity(1024);
+    for resized in &[false, true] {
+        let mut actual_size: u32 = 0;
+        let status = OakStatus::from_i32(unsafe {
+            oak_abi::channel_label(
+                handle,
+                bytes.as_mut_ptr(),
+                bytes.capacity(),
+                &mut actual_size,
+            ) as i32
+        });
+        match status {
+            Some(s) => match s {
+                OakStatus::Ok => {
+                    unsafe {
+                        // The serialized label was successfully fetched, so set the length to the
+                        // actual length.
+                        bytes.set_len(actual_size as usize);
+                    }
+                    return Ok(Label::deserialize(&bytes).expect("Could not deserialize label."));
+                }
+
+                OakStatus::ErrBufferTooSmall if !(*resized) => {
+                    // Extend the vector to be large enough for the serialized label.
+                    debug!(
+                        "Got space for {} bytes, need {}",
+                        bytes.capacity(),
+                        actual_size
+                    );
+                    bytes.reserve((actual_size as usize) - bytes.capacity());
+
+                    // Try again with a buffer resized to cope with expected size of data.
+                    continue;
+                }
+
+                s => {
+                    return Err(s);
+                }
+            },
+            None => {
+                return Err(OakStatus::ErrInternal);
+            }
+        }
+    }
+    error!("unreachable code reached");
+    Err(OakStatus::ErrInternal)
+}
+
+/// Get the [`Label`] for the current node.
+pub fn node_label() -> Result<Label, OakStatus> {
+    let mut bytes = Vec::with_capacity(1024);
+    for resized in &[false, true] {
+        let mut actual_size: u32 = 0;
+        let status = OakStatus::from_i32(unsafe {
+            oak_abi::node_label(bytes.as_mut_ptr(), bytes.capacity(), &mut actual_size) as i32
```

Comment:
Again, it would be good to reduce the copy-pasta and have a single helper used in 3 places; this one will be a bit more complex though (pass in a lambda as a `FnOnce` maybe?).

Reply:
Done, except used `Fn` rather than `FnOnce` as it could be called twice if the label is larger than 1024 bytes.